lib/pure/httpclient.nim

```diff
@@ -930,7 +930,8 @@ proc newConnection(client: HttpClient | AsyncHttpClient,
       else: nativesockets.Port(connectionUrl.port.parseInt)
 
     when client is HttpClient:
-      client.socket = await net.dial(connectionUrl.hostname, port)
+      client.socket = newSocket()
+      client.socket.connect(connectionUrl.hostname, port, client.timeout)
```

Comment:
This will prevent anything else from running. Isn't there a way to initiate the connection, and wait until either the connection finished, or a timeout occurs (whichever comes first)?

Reply:
Hi, I don't understand, this is a blocking not async variant. Not sure what you mean by `prevent from running`. What is currently happening is that the timeout passed in is not used and so the blocking variant will indeed timeout after some time but according to a timeout it is called with.

My cnowledge of Nim is limited but I am using a preflight check with a socket.connect just because `timeout` variable is not working. I would like to have it fixed somehow.

Reply:
What I mean is that, I believe this will block the entire thread. Ideally this should run in such a way that it doesn't prevent other scheduled asynchronous routines from running.

Reply:
@Varriount as viralpoetry said, and if you look at the code, it only changes the sync code path, because timeouts for async already work

Reply:
await in this case means nothing, I think it was used here as a mistake as a part of multisync